R/hamcrest.R

```diff
@@ -1,162 +1,341 @@
-  nonFiniteValuesIdentical <- identical(expected[!finite], actual[!finite])
-  
-  return( finiteValuesCloseEnough && nonFiniteValuesIdentical )
-}
-
-identical.attributes <- function(actual, expected, tol = NULL) {
-    # Should have the same set of names,
-    # though not necessarily in the same order
-    if(length(setdiff(names(expected), names(actual))) > 0) {
-        return(FALSE)
-    }
-    
-    # Otherwise verify that the values are identical
-    for(a in names(expected)) {
-        if(!identical.rec(actual[[a]], expected[[a]], tol)) {
-            return(FALSE)
-        }
-    }
-    return(TRUE)
-}
-
-identical.rec <- function(actual, expected, tol = NULL) {
-    if (length(actual) != length(expected))
-      return(FALSE)
-    if (typeof(actual) != typeof(expected))
-      return(FALSE)
-    if (!identical.attributes(attributes(actual), attributes(expected), tol)) {
-      return(FALSE)
-    }
-    if (is.list(actual)) {
-      for (i in seq_along(actual)) {
-        isSame <- identical.rec(actual[[i]], expected[[i]], tol)
-        if (!isSame){
-          return(FALSE)
-        }
-      }
-      return(TRUE)
-    } else if (!is.null(tol) && is.double(actual)) {
-      compareReal(unclass(actual), unclass(expected), tol)
-    } else if (!is.null(tol) && is.complex(actual)) {
-      compareReal(unclass(Re(actual)), unclass(Re(expected)), tol) &&
-        compareReal(unclass(Im(actual)), unclass(Im(expected)), tol)
-    } else {
-      return(identical(actual, expected))
-    }
+	}
 }
 
+### ----------------------------------------------------------------- ###
+### MATCHERS ----
+### ----------------------------------------------------------------- ###
 
+#' Checks if the expected value is close to a value within some tolerance level
+#'
+#' A \emph{matcher} function which will return \code{TRUE} if its first argument
+#' is a numeric vector of length 1 whose absolute difference with expected is
+#' greater than or equal to the value of \code{delta}.
+#'
+#' @param expected a numeric vector.
+#' @param delta a numeric vector of length one that defines the maximum allowed
+#'   difference.
+#' @details
+#' The \code{closeTo()} matcher function can only be used for numeric arguments.
+#' @examples \dontrun{
+#' assertThat(-0.50557992900139, closeTo(-0.50557, delta = 1e4))
+#' }
+#' @seealso
+#' \itemize{
+#' \item \code{\link{identicalTo}}
+#' \item \code{\link{equalTo}}
+#' }
+#' @export
 closeTo <- function(expected, delta) {
     stopifnot(is.numeric(expected) & is.numeric(delta) & length(delta) == 1L)
 	function(actual) {
 		length(expected) == length(actual) &&
-				all(abs(expected-actual)<delta)	
+				all(abs(expected-actual)<delta)
 	}
 }
 
+#' Checks if the expected value is identical to a value
+#'
+#' A \emph{matcher} function which will return \code{TRUE} if the actual value
+#' is identical to the expected value.
+#'
+#' @param expected object passed to the matcher function.
+#' @param tol numeric tolerance.
```

Comment:
Should not that `tol` is only used if the expected value is a finite floating point value. If the expected value is another type, such as an integer, an exact match is expected.

Reply:
For me, the argument `tol` does not suit well with `identicalTo()` as that tolerance level makes more sense with `equalTo()`. Users testing floating point values should rather use `equalTo()`. And, `identicalTo()` should be used when they test finite or categorical values e.g. integers, characters and so on; therefore, a tolerance level in there doesn't make any sense. 

What do you think about removing `tol` argument from `identicalTo()`?